File: 682-baseball-game/682-baseball-game.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        l = list()
        
        for i in range(len(ops)):
            if( ops[i] == "D"):                
                n = l[len(l) - 1]
                l.append(n * 2)
                # print(l)
            elif(ops[i] == "C"):
                l.pop()
                # print(l)
                
            elif(ops[i] == "+"):
                # print(l)
                n1 = l[len(l) - 1]
                n2 = l[len(l) - 2]
                
                l.append(n1 + n2)
                
            else:
                
                l.append(int(ops[i]))
                # print(l)
                
        ans = 0
        for i in l:
            ans+=i
            
        return ans
        

File: 682-baseball-game/682-baseball-game.cpp
#include <cstdlib>
#include <string>

class Solution {
public:
    int calPoints(vector<string>& ops) {
        
        int n = ops.size();
        vector<int> ans;
        for(int i = 0 ; i < n ; i++)
        {
            string temp = ops[i];
            
            if(temp == "C"){
                
                ans.pop_back();
            }
            
            else if(temp == "D"){
                int num = ans[ans.size() -1];
                // ans.pop_back();
                ans.push_back(num * 2);
                // ops.pop_back();
                // ops.push_back(num * 2); --> here is the error
                //  it is vecotr of strings but i am pushing int adeke not working
             }
            
            else if(temp == "+"){
                int n1 = ans[ans.size() - 1];
                int n2 = ans[ans.size() - 2];
                // ans.pop_back();
                // ans.pop_back();                
                ans.push_back(n1 + n2);
            }
            
            else{
                ans.push_back(stoi(temp));
            }
        }
        int ansSum = 0;
        
        for(int i = 0 ; i < ans.size() ;i++){
            ansSum += ans[i];
        }
        return ansSum;
        
    }
};